src/libpriv/rpmostree-core.cxx

```diff
@@ -3786,10 +3786,9 @@ write_rpmdb (RpmOstreeContext      *self,
    * to find the rpmdb and RPM macros are global state. */
   set_rpm_macro_define ("_dbpath", "/" RPMOSTREE_RPMDB_LOCATION);
 
-  if (self->treefile_rs && !self->treefile_rs->rpmdb_backend_is_default())
+  if (self->treefile_rs && self->treefile_rs->rpmdb_backend_is_target())
     {
-      auto rpmdb_type = self->treefile_rs->get_rpmdb();
-      g_print ("Regenerating rpmdb: %s\n", rpmdb_type.c_str());
+      g_print ("Regenerating rpmdb for target\n");
       rpmostreecxx::rewrite_rpmdb_for_target(tmprootfs_dfd);
```

Comment:
Hmm, so now we'll be calling this much more often, even if the rpmdb target and host match, right? I guess we can optimize this down the road.

Edit: I think we'd just need to check if `'%{_db_backend}'` is the same value in both the host and the target.

Reply:
> Edit: I think we'd just need to check if '%{_db_backend}' is the same value in both the host and the target.

Yeah, seems a bit tricky though.